Week2/app.js

```diff
@@ -0,0 +1,172 @@
+             if (result2[0]=== undefined) {
+                 console.log(`${input2.Region} is not exist in database, please insert a valid value`);
+                 
+             } else {
+                 console.log(`The spoken languages in ${input2.Region} are:`);
+                 console.table(result2)
+             }
+// 3- Find the number of cities in which language Z is spoken (Accept Z from user)
+     } else if (question[0] === 'number of cities'){
+        const input6 = await inquirer.prompt([
+                { name: 'language', message: 'Insert language:' },
+                ]);
+        const sql6 = `SELECT  count(city.name) as Cities FROM city join countrylanguage on city.CountryCode = countrylanguage.CountryCode where Language=? `;
+                        let result6 = await queryPromise(sql6, input6.language);
+                        if (result6[0].Cities === 0) {
+                            console.log(`${input6.language} is not exist in database, please insert a valid value`);
+                         
+                        } else {
+                            console.log(`The number of cities which speaking in ${input6.language} is: ${result6[0].Cities}`);
+                        }
+    // 4- Accept the region and language from the user. Are there any countries in this region with the given language as the official language 
+     
+     } else if(question[0] === 'official language'){
+        const input4 = await inquirer.prompt([
+             { name: 'Region', message: 'Insert Region:' },
+             { name: 'language', message: 'Insert language:' },
+        ]);
+        const sql4 = `SELECT Name as Country FROM country join countrylanguage on country.Code=countrylanguage.CountryCode where Region =? and Language=? and IsOfficial='T'; `;
+        let result4 = await queryPromise(sql4, [input4.Region, input4.language,input4.official]);
+            if (result4[0] === 0 ) {
+                console.log(`${input4.language} is not official language in any country in ${input4.Region} `);
+            } else {
+                console.log(`${input4.language} is official language in:`);
+                console.table(result4)
+                           
+            }
+    // 5- List all the continents with the number of languages spoken in each continent
+    }else if (question[0] === 'languages in continent' ) {
+                    const sql5 = `SELECT Continent, COUNT(DISTINCT Language) as Languages FROM country join countrylanguage on country.Code=countrylanguage.CountryCode GROUP BY Continent`;
+                    let result5 = await queryPromise(sql5);
+                    console.table(result5);
+    // 6- Get alerts when I want to insert row if a country has >= 10 languages.
+
+} else if (question[0] === 'insert row'){
+   
+       const  sql6 = 'SELECT  name as Country, COUNT(Language) as Number_of_Languages FROM countrylanguage join country on  countrylanguage.CountryCode = country.Code GROUP BY name'
```

Comment:
It's a nice idea and this actually works if you are just using your entry point to the program. But what if I want to prevent EVERYONE to enter more than 10 languages? Meaning neither by the mysql console or my program or anything that could connect to my db ? 
The solution would be to create the trigger. It is a small piece of sql run before or after an insert/update/delete and that can prevent you from doing actions.
I invite you to have a look on it and if you want try to do the same thing using a mysql trigger.

Reply:
I have never thought about TRIGGER when I looked at the home work. 
I changed the code now using trigger, It was really interesting.